src/probnum/problems/zoo/filtsmooth/_filtsmooth_problems.py

```diff
@@ -0,0 +1,470 @@
+        y2 = x2 - g * np.sin(x1) * step
+        return np.array([y1, y2])
+
+    def df(t, x):
+        x1, x2 = x
+        y1 = [1, step]
+        y2 = [-g * np.cos(x1) * step, 1]
+        return np.array([y1, y2])
+
+    def h(t, x):
+        x1, x2 = x
+        return np.array([np.sin(x1)])
+
+    def dh(t, x):
+        x1, x2 = x
+        return np.array([[np.cos(x1), 0.0]])
+
+    process_noise_cov = (
+        np.diag(np.array([step ** 3 / 3, step]))
+        + np.diag(np.array([step ** 2 / 2]), 1)
+        + np.diag(np.array([step ** 2 / 2]), -1)
+    )
+
+    dynamics_model = statespace.DiscreteGaussian(
+        input_dim=2,
+        output_dim=2,
+        state_trans_fun=f,
+        proc_noise_cov_mat_fun=lambda t: process_noise_cov,
+        jacob_state_trans_fun=df,
+    )
+
+    measurement_model = statespace.DiscreteGaussian(
+        input_dim=2,
+        output_dim=1,
+        state_trans_fun=h,
+        proc_noise_cov_mat_fun=lambda t: measurement_variance * np.eye(1),
+        jacob_state_trans_fun=dh,
+    )
+
+    if initrv is None:
+        initrv = randvars.Normal(np.ones(2), measurement_variance * np.eye(2))
+
+    # Generate data
+    times = np.arange(0.0, t_max, step=step)
+    states, obs = statespace.generate_samples(
+        dynmod=dynamics_model, measmod=measurement_model, initrv=initrv, times=times
+    )
+    regression_problem = problems.RegressionProblem(
+        observations=obs, locations=times, solution=states
+    )
+    statespace_components = dict(
+        dynamics_model=dynamics_model,
+        measurement_model=measurement_model,
+        initrv=initrv,
+    )
+    return regression_problem, statespace_components
+
+
+def benes_daum(
+    measurement_variance: FloatArgType = 0.1,
+    process_diffusion: FloatArgType = 1.0,
+    initrv: Optional[randvars.RandomVariable] = None,
+):
+    r"""Filtering/smoothing setup based on the Benes SDE.
+
+    A non-linear state space model for the dynamics of a Benes SDE.
+    Here, we formulate a continuous-discrete state space model:
+
+    .. math::
+
+        x(t_n) &= \tanh(x(t_{n-1})) + q(t_n) \\
+        y_n &= x(t_n) + r_n
+
+    for some scalar :math:`\lambda` and :math:`q(t_n) \sim \mathcal{N}(0, Q)`,
+    :math:`r_n \sim \mathcal{N}(0, R)` for process noise covariance matrix :math:`Q`
+    and measurement noise covariance matrix :math:`R`.
+
+    Parameters
+    ----------
+    measurement_variance
+        Marginal measurement variance.
+    process_diffusion
+        Diffusion constant for the dynamics
+    times
+        Time grid for the filtering/smoothing problem.
+    initrv
+        Initial random variable.
+
+    Returns
+    -------
+    statespace_components
+        Dictionary containing
+        - dynamics model
+        - measurement model
+        - initial random variable
+
+    Notes
+    -----
+    No ``RegressionProblem`` object is returned. The dynamics model has to be linearized
+    in order to generate data.
```

Comment:
hmmm this is unfortunate. Maybe we should implement some simple Euler-Maruyama scheme for non-linear SDEs, in order to solve problems like this one.

In any case, do you think it would be more clear if the output of this function would still be a tuple, e.g. `None, {...}` instead of only the dict? This way all the zoo problems would have similar signatures. What do you think?